### DISCUSSION TITLE
Eliminate duplicate content from context

### DISCUSSION THREAD

**Discussion by napter** [cline/cline#748] [Feature Requests]
When the model reads a lot of files - or does a lot of diffs, the context can become really large and can contain a lot of duplicate content - files read multiple times, diffs with small changes, etc. A way to remove duplicate content before submitting would save time and tokens.

**Comment by jakenuts** [comment]
Context management is definitely a key area to improve Cline's effectiveness. Apart from the obvious problems "it read 10,000 line output and 💥" there's also focus loss as the context grows. 

I'd welcome an extension point where any particular output was about to be added to the context (human input too) to make a context management system easier to add on. 

A more advanced implementation would support curation, deduplication, trimming the context as needed. I know that would break prompt caching so it would need to be carefully managed but it would definitely improve its output. 

I am starting to think that a formal workflow (defined in a prompt and some local files) is the key to getting the most out of Cline. Like rules for what to read, what to remember, how to record & update tasks, and most importantly the "handoff" to a new Cline instance. I've heard others say they get better output if they use a new instance for every task, the key being how to keep it on track and not reproduce earlier effort. 

Finally I've read that having false paths (syntax errors that were later fixed) in the context reduces the agents effectiveness. So instead of a whole section where it flailed you could reduce that to a single instruction "method x is not available use y" sort of thing. (ps not "don't try x" as that's also considered demotivating)
